code/game/machinery/computer/shuttle_computers.dm

```diff
@@ -1,3 +1,48 @@
+//Docking port disks
+//Insert into a shuttle computer to unlock a new destination
+/obj/item/weapon/disk/shuttle_coords
+	name = "shuttle destination disk"
+	desc = "A small disk containing encrypted coordinates and tracking data."
+	icon = 'icons/obj/cloning.dmi'
+	icon_state = "datadisk0"
+
+	var/obj/docking_port/destination/destination //Docking port linked to this disk.
+	//If this variable contains a path like (/obj/structure/docking_port/destination/my_dungeon), the disk will find a destination docking port of that type and automatically link to it
+	//See example below
+
+	var/header = "SDC Data Disk" //Name of the disk, shown on the console. SDC stands Shuttle Destination Coordinates
+
+	var/list/allowed_shuttles = list() //List of allowed shuttles. Accepts paths (for example /datum/shuttle/arrival). If empty, all shuttles are allowed
+
+//Example:
+/obj/item/weapon/disk/shuttle_coords/station_arrivals
+	destination = /obj/docking_port/destination/transport/station
+	header = "station arrivals"
+
+/obj/item/weapon/disk/shuttle_coords/vault
+	allowed_shuttles = list(/datum/shuttle/mining, /datum/shuttle/research)
+
+//This disk will link to station's arrivals when spawned
+
+/obj/item/weapon/disk/shuttle_coords/New()
+	..()
+
+	if(ispath(destination))
+		spawn()
```

Comment:
No sense in spawning this, the locate will not be particularly expensive, and you'd never want to be using a shuttle disk without its destination set anyways.